[daint-mc dom-mc] adding a new recipe for daint-mc ParaView5.12

jfavre opened the pull request.

jfavre commented:
retest this please

jfavre commented:
the build fails with ospray, but my own testing shows that it works fine using my personal account. Can @lucamar make a suggestion?

jfavre@daint104:~/production/easybuild/easyconfigs/o/ospray> eb -f ospray-2.12.0-CrayGNU-21.09.eb -r
== Temporary log file in case of crash /run/user/1100/easybuild/tmp/eb-3xsv4zbr/easybuild-zlverruy.log
== resolving dependencies ...
== processing EasyBuild easyconfig /users/jfavre/production/easybuild/easyconfigs/o/ospray/ospray-2.12.0-CrayGNU-21.09.eb
== building and installing ospray/2.12.0-CrayGNU-21.09...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
== ... (took 2 secs)
== building...
== ... (took 4 mins 44 secs)
== testing...
== installing [skipped]
== taking care of extensions...
== restore after iterating...
== postprocessing...
== sanity checking...
== cleaning up...
== creating module...
== permissions...
== packaging...
== COMPLETED: Installation ended successfully (took 4 mins 50 secs)
== Results of the build can be found in the log file(s) /scratch/snx3000/jfavre/daint/software/ospray/2.12.0-CrayGNU-21.09/easybuild/easybuild-ospray-2.12.0-20240109.082823.log
== Build succeeded for 1 out of 1

lucamar commented:
As far as I can read in the log file `/scratch/snx3000/jenscscs/TestingEB/daint-mc/44dfdad-6408/tmp/eb-gz87l687/easybuild-ospray-2.12.0-20240109.130644.VSUjd.log` on the system, the error is related to Git LFS:
```
Error: The file "/tmp/jenscscs/44dfdadmc/ospray/2.12.0/CrayGNU-21.09/easybuild_obj/oidn/src/weights/rt_ldr.tza" is a Git LFS pointer. Please install Git LFS and clone the repository again.
Error: The file "/tmp/jenscscs/44dfdadmc/ospray/2.12.0/CrayGNU-21.09/easybuild_obj/oidn/src/weights/rt_hdr_alb.tza" is a Git LFS pointer. Please install Git LFS and clone the repository again.
Error: The file "/tmp/jenscscs/44dfdadmc/ospray/2.12.0/CrayGNU-21.09/easybuild_obj/oidn/src/weights/rt_hdr_alb_nrm.tza" is a Git LFS pointer. Please install Git LFS and clone the repository again.
```

jfavre commented:
> As far as I can read in the log file `/scratch/snx3000/jenscscs/TestingEB/daint-mc/44dfdad-6408/tmp/eb-gz87l687/easybuild-ospray-2.12.0-20240109.130644.VSUjd.log` on the system, the error is related to Git LFS:
> 
> ```
> Error: The file "/tmp/jenscscs/44dfdadmc/ospray/2.12.0/CrayGNU-21.09/easybuild_obj/oidn/src/weights/rt_ldr.tza" is a Git LFS pointer. Please install Git LFS and clone the repository again.
> Error: The file "/tmp/jenscscs/44dfdadmc/ospray/2.12.0/CrayGNU-21.09/easybuild_obj/oidn/src/weights/rt_hdr_alb.tza" is a Git LFS pointer. Please install Git LFS and clone the repository again.
> Error: The file "/tmp/jenscscs/44dfdadmc/ospray/2.12.0/CrayGNU-21.09/easybuild_obj/oidn/src/weights/rt_hdr_alb_nrm.tza" is a Git LFS pointer. Please install Git LFS and clone the repository again.
> ```

Thanks, I had overlooked this error. Still, I don't understand it since git-lfs is installed in /scratch/snx3000/jenscscs/TestingEB/daint-mc/71adad2-6409/software/git-lfs/3.2.0/bin

jfavre commented:
I rebuild ospray with the newer git-lfs and was successful:
== COMPLETED: Installation ended successfully (took 4 mins 43 secs)
== Results of the build can be found in the log file(s) /scratch/snx3000tds/jfavre/daint/software/ospray/2.12.0-CrayGNU-21.09/easybuild/easybuild-ospray-2.12.0-20240111.182136.log

lucamar commented on jenkins-builds/7.0.UP03-21.09-daint-mc:
```diff
@@ -7,6 +7,7 @@
  CP2K-9.1-CrayGNU-21.09.eb                              --set-default-module
  CPMD-4.3-CrayIntel-21.09.eb                            --set-default-module
  FFmpeg-5.0-CrayGNU-21.09.eb                            --set-default-module
+ git-lfs-3.2.0.eb                                       --set-default-module
```
```suggestion
git-lfs-3.2.0.eb                                       --set-default-module
git-lfs-3.4.1.eb
```

lucamar reviewed (approved):
Thanks @jfavre: I suggest to update `git-lfs` in the `jenkins-builds` list and merge the request as it is, since it seems that `ospray` with `git-lfs` cannot be installed easily on the Lustre file system, while it succeeds on GPFS.

lucamar commented:
I have added `prebuildopts += "git lfs install && "` to the `ospray` recipe, which is needed to initialize`git lfs`. I will add the initialization command as well to the recipe `git-lfs-3.4.1.eb`, in order to ensure that `git lfs` is initialized when the module `git-lfs` is loaded: anyway it does not harm to keep the command `git lfs install` in the `ospray` recipe.

jfavre commented:
> I have added `prebuildopts += "git lfs install && "` to the `ospray` recipe, which is needed to initialize`git lfs`. I will add the initialization command as well to the recipe `git-lfs-3.4.1.eb`, in order to ensure that `git lfs` is initialized when the module `git-lfs` is loaded: anyway it does not harm to keep the command `git lfs install` in the `ospray` recipe.

I was testing the same install on LUMI and had the same error. Once I added your magic patch, things worked on LUMI as well. Bravo @lucamar !
